pkg/conference/peer_message_processor.go

```diff
@@ -68,40 +67,55 @@ func (c *Conference) processICEGatheringCompleteMessage(participant *Participant
 
 func (c *Conference) processRenegotiationRequiredMessage(participant *Participant, msg peer.RenegotiationRequired) {
 	participant.logger.Info("Started renegotiation")
-	participant.sendDataChannelMessage(event.SFUMessage{
-		Op:       event.SFUOperationOffer,
-		SDP:      msg.Offer.SDP,
-		Metadata: c.getAvailableStreamsFor(participant.id),
+	participant.sendDataChannelMessage(event.Event{
+		Type: event.FocusCallNegotiate,
+		Content: event.Content{
+			Parsed: event.FocusCallNegotiateEventContent{
+				Description: event.CallData{
+					Type: event.CallDataType(msg.Offer.Type.String()),
+					SDP:  msg.Offer.SDP,
+				},
+				SDPStreamMetadata: c.getAvailableStreamsFor(participant.id),
+			},
+		},
 	})
 }
 
 func (c *Conference) processDataChannelMessage(participant *Participant, msg peer.DataChannelMessage) {
 	participant.logger.Debug("Received data channel message")
-	var sfuMessage event.SFUMessage
-	if err := json.Unmarshal([]byte(msg.Message), &sfuMessage); err != nil {
+	var focusEvent event.Event
+	if err := focusEvent.UnmarshalJSON([]byte(msg.Message)); err != nil {
 		c.logger.Errorf("Failed to unmarshal SFU message: %v", err)
 		return
 	}
 
-	switch sfuMessage.Op {
-	case event.SFUOperationSelect:
-		c.processSelectDCMessage(participant, sfuMessage)
-	case event.SFUOperationAnswer:
-		c.processAnswerDCMessage(participant, sfuMessage)
-	case event.SFUOperationPublish, event.SFUOperationUnpublish:
-		c.processPublishDCMessage(participant, sfuMessage)
-	case event.SFUOperationAlive:
-		c.processAliveDCMessage(participant)
-	case event.SFUOperationMetadata:
-		c.processMetadataDCMessage(participant, sfuMessage)
+	switch focusEvent.Type.Type {
+	case event.FocusCallTrackSubscription.Type:
+		focusEvent.Content.ParseRaw(event.FocusCallTrackSubscription)
```

Comment:
Just a question: is there a way we could avoid calling `ParseRaw()` for every event type?

I.e. so that it's enough to call `AsXXX()` after we checked the `EventType`? (similar to how we do it with other signaling messages from e.g. Matrix SDK, where we just check the type and then call one of the `AsXXX()` functions?)